src/analysis/fourier/poisson_summation.lean

```diff
@@ -112,3 +123,140 @@ begin
     rw [←real.fourier_coeff_tsum_comp_add h_norm n, fourier_eval_zero, smul_eq_mul, mul_one],
     refl },
 end
+
+section rpow_decay
+
+variables {E : Type*} [normed_add_comm_group E]
+
+/-- If `f` is `O(x ^ (-b))` at infinity, then so is the function
+`λ x, ‖f.restrict (Icc (x + R) (x + S))‖` for any fixed `R` and `S`. -/
+lemma is_O_norm_Icc_restrict_at_top {f : C(ℝ, E)} {b : ℝ} (hb : 0 < b)
+  (hf : is_O at_top f (λ x : ℝ, |x| ^ (-b))) (R S : ℝ) :
+  is_O at_top (λ x : ℝ, ‖f.restrict (Icc (x + R) (x + S))‖) (λ x : ℝ, |x| ^ (-b)) :=
+begin
+  -- First establish an explicit estimate on decay of inverse powers.
+  -- This is logically independent of the rest of the proof, but of no mathematical interest in
+  -- itself, so it is proved using `async` rather than being formulated as a separate lemma.
+  have claim : ∀ (x : ℝ), max 0 (-2 * R) < x →
+    ∀ (y : ℝ), x + R ≤ y → y ^ (-b) ≤ (1 / 2) ^ (-b) * x ^ (-b),
+  async { intros x hx y hy,
```

Comment:
```suggestion
  { intros x hx y hy,
```
This is not really used in mathlib, and I don't know how portable it is to Lean 4, so let's avoid it for now. In any case, `have` claims would be automatically proved asynchronously in a perfect world.